dev/cnf/gradle/scripts/tasks.gradle

```diff
@@ -14,6 +14,7 @@ subprojects {
 
   plugins.apply 'java'
   plugins.apply 'maven-publish'
+  plugins.apply 'jacoco'
```

Comment:
lets move all javacoco specific stuff into a big if statement, so that we only apply the jacoco plugin and tasks if we are actually going to run them

Reply:
After talking to Andy about the above comment, we decided to leave this where it is. The configuration I added below will prevent the Jacoco Java Agent from attaching to the JUnit test process and collecting stats. Since the jacocoTestReport and jacocoTestCoverageVerification tasks are Jacoco plugin tasks, this will also keep the tasks in the Gradle tasks lists instead of being hidden.